pkg/component/etcd/configmap/configmap.go

```diff
@@ -57,106 +63,86 @@ func (c *component) Deploy(ctx context.Context) error {
 	return nil
 }
 
-func (c *component) syncConfigmap(ctx context.Context, cm *corev1.ConfigMap) error {
-	etcdConfigYaml := `# Human-readable name for this member.
-    name: ` + fmt.Sprintf("etcd-%s", c.values.EtcdUID[:6]) + `
-    # Path to the data directory.
-    data-dir: /var/etcd/data/new.etcd`
-	metricsLevel := druidv1alpha1.Basic
-	if c.values.Metrics != nil {
-		metricsLevel = *c.values.Metrics
-	}
-
-	etcdConfigYaml = etcdConfigYaml + `
-    # metrics configuration
-    metrics: ` + string(metricsLevel) + `
-    # Number of committed transactions to trigger a snapshot to disk.
-    snapshot-count: 75000
-
-    # Accept etcd V2 client requests
-    enable-v2: false`
-	var quota int64 = 8 * 1024 * 1024 * 1024 // 8Gi
-	if c.values.Quota != nil {
-		quota = c.values.Quota.Value()
-	}
-
-	etcdConfigYaml = etcdConfigYaml + `
-    # Raise alarms when backend size exceeds the given quota. 0 means use the
-    # default quota.
-    quota-backend-bytes: ` + fmt.Sprint(quota)
-
-	enableTLS := (c.values.ClientUrlTLS != nil)
-
-	protocol := "http"
-	if enableTLS {
+func generateTLSConfig(serverType, namespace, svcName, port string, tlsConfig *druidv1alpha1.TLSConfig) (etcdConfigYaml string) {
+	var protocol = "http"
+	if tlsConfig != nil {
 		protocol = "https"
 		tlsCASecretKey := "ca.crt"
-		if dataKey := c.values.ClientUrlTLS.TLSCASecretRef.DataKey; dataKey != nil {
+		if dataKey := tlsConfig.TLSCASecretRef.DataKey; dataKey != nil {
 			tlsCASecretKey = *dataKey
 		}
-		etcdConfigYaml = etcdConfigYaml + `
-    client-transport-security:
-        # Path to the client server TLS cert file.
-        cert-file: /var/etcd/ssl/client/server/tls.crt
+		etcdConfigYaml = `
+    ` + serverType + `-transport-security:
+        # Path to the ` + serverType + ` server TLS cert file.
+        cert-file: /var/etcd/ssl/` + serverType + `/server/tls.crt
 
-        # Path to the client server TLS key file.
-        key-file: /var/etcd/ssl/client/server/tls.key
+        # Path to the ` + serverType + ` server TLS key file.
+        key-file: /var/etcd/ssl/` + serverType + `/server/tls.key
 
-        # Enable client cert authentication.
+        # Enable ` + serverType + ` cert authentication.
         client-cert-auth: true
 
-        # Path to the client server TLS trusted CA cert file.
-        trusted-ca-file: /var/etcd/ssl/client/ca/` + tlsCASecretKey + `
+        # Path to the ` + serverType + ` server TLS trusted CA cert file.
+        trusted-ca-file: /var/etcd/ssl/` + serverType + `/ca/` + tlsCASecretKey + `
 
-        # Client TLS using generated certificates
+        # ` + strings.Title(serverType) + ` TLS using generated certificates
         auto-tls: false`
```

Comment:
From [here](https://etcd.io/docs/v3.2/op-guide/security/#basic-setup) seems like its an option to use automatically generated certs and we don’t need that since we provide certs